File: projects/models/project_style.py
from django.db import models

from libraries.models import Dictable


class ProjectStyle(Dictable):

    name = models.CharField(max_length=255)
    iconClass = models.CharField(max_length=255)
    arrowClass = models.CharField(max_length=255)
    lineClasses = models.CharField(max_length=255)

    def __str__(self):
        return self.name

File: static/js/models/project-style.js
define(
    [
        'models/object-configurable-class',
        'project-constants'
    ],
    function(
        ObjectConfigurable,
        projectConstants
    ) {

        /**
         * Class for storing information about a project's style
         * @constructor
         */
        function ProjectStyle(configuration) {

            this.iconClass = null;
            this.arrowClass = null;
            this.lineClasses = [];

            ObjectConfigurable.call(this, configuration);
            this.lineClasses = this.lineClasses.split(',');

        }

        ProjectStyle.prototype.randomize = function() {

        };

        return ProjectStyle;

    }
);
